code/datums/components/orbiter.dm

```diff
@@ -89,19 +89,45 @@
 	to_chat(orbiter, "<span class='notice'>Now orbiting [parent].</span>")
 
 /datum/component/orbiter/proc/end_orbit(atom/movable/orbiter, refreshing=FALSE)
-	if(!orbiters[orbiter])
+	if(!current_orbiters[orbiter])
 		return
 	UnregisterSignal(orbiter, COMSIG_MOVABLE_MOVED)
 	SEND_SIGNAL(parent, COMSIG_ATOM_ORBIT_STOP, orbiter)
 	orbiter.SpinAnimation(0, 0)
-	if(istype(orbiters[orbiter],/matrix)) //This is ugly.
-		orbiter.transform = orbiters[orbiter]
-	orbiters -= orbiter
+	if(istype(current_orbiters[orbiter],/matrix)) //This is ugly.
+		orbiter.transform = current_orbiters[orbiter]
+	current_orbiters -= orbiter
 	orbiter.stop_orbit(src)
 	orbiter.orbiting = null
-	if(!refreshing && !length(orbiters) && !QDELING(src))
+	if(!refreshing && !length(current_orbiters) && !QDELING(src))
 		qdel(src)
 
+/**
+ * [Proc Behavior]
+ * 		If target_orbited is null, incoming_orbiter will stop orbiting and start to orbit the new target.
+ * 			-> This is because 'check_orbitable()' makes an orbit component properly.
+ * 		If original_orbited.current_orbiters has one orbiter, incoming_orbiter will stop orbiting and start to orbit the new target.
+ * 			-> This is beacuse 'end_orbit()' removes an orbit component properly.
+ * 		If target_orbited is not null, manually control signals (send, register, unregister), then transfer to the new target.
+ * 			-> We can keep ghosts' orbit animation without glitching theirs.
+ */
+/datum/component/orbiter/proc/transfer_orbiter_to(atom/movable/incoming_orbiter, atom/new_target)
+	if(!new_target || !incoming_orbiter)
+		return
+	if(!new_target.orbit_datum || length(current_orbiters)==1) // if target has no orbiters or original orbit has only one orbiter
+		end_orbit(incoming_orbiter)
+		incoming_orbiter.check_orbitable(new_target)
+		return
+
+	SEND_SIGNAL(parent, COMSIG_ATOM_ORBIT_STOP, incoming_orbiter)
+	UnregisterSignal(incoming_orbiter, COMSIG_MOVABLE_MOVED)
+	new_target.orbit_datum.RegisterSignal(incoming_orbiter, COMSIG_MOVABLE_MOVED, PROC_REF(orbiter_move_react))
```

Comment:
Where is this signal unregistered? Also if the orbit_datum is a datum, you cannot register a moved signal to it as that only applies to atom/movable

Reply:
Why is this line even needed?, if it was working in your testing then it doens't seem necessary to me since this shouldn't work anyway

Reply:
I am not the one who can answer why it's necessary for the code. What I did is only copy-paste the behaviour since it's transferring orbiter actually needs the reapplication of the orbit applying code.